core/src/server/rpc_module.rs

```diff
@@ -150,9 +211,48 @@ pub enum MethodCallback {
 	/// Asynchronous method handler.
 	Async(AsyncMethod<'static>),
 	/// Subscription method handler.
-	Subscription(SubscriptionMethod<'static>),
+	Subscription {
+		/// The subscription method handler.
+		method: SubscriptionMethod<'static>,
+		/// The name of the associated unsubscription method.
+		unsubscribe_method_name: &'static str,
+	},
 	/// Unsubscription method handler.
-	Unsubscription(UnsubscriptionMethod),
+	Unsubscription {
+		/// The unsubscription method handler, usually auto-generated on
+		/// subscription insertion.
+		method: UnsubscriptionMethod,
+		/// The name of the associated subscription method.
+		subscribe_method_name: &'static str,
+	},
+	/// The method is an alias for another method.
+	Alias(WeakMethodCallback<'static>),
```

Comment:
How is this intended to work?
If one enables an alias for method a and method a is replaced by another callback?
Is the alias just disabled i.e, dangling/weak or is supposed to point the new callback?

Reply:
> If one enables an alias for method a and method a is replaced by another callback?

like the current aliasing system, the alias points to the underlying callback, not to the aliased name. However, unlike the current aliasing system, the alias no longer keeps the callback "alive" like the canonical name does. Removing the canonical name will remove the callback from the map and the alias's will no longer work

Using a weak is important to preventing dangling handlers. If we allow the alias to keep the callback alive, then we can create dangling handlers by aliasing subscribe functions before removing them